--- Modules/Setup.in
+++ Modules/Setup.in
@@ -84,7 +84,7 @@
 fcntl fcntlmodule.c	# fcntl(2) and ioctl(2)
 pwd pwdmodule.c		# pwd(3)
 grp grpmodule.c		# grp(3)
-crypt cryptmodule.c	# crypt(3)
+crypt cryptmodule.c -lcrypt	# crypt(3)
 select selectmodule.c	# select(2); not on ancient System V
 socket socketmodule.c	# socket(2); not on ancient System V
 signal signalmodule.c	# signal(2)

--- Objects/fileobject.c
+++ Objects/fileobject.c
@@ -361,7 +361,7 @@
 */

 static object *
-getline(f, n)
+mygetline(f, n)
 	fileobject *f;
 	int n;
 {
@@ -482,7 +482,7 @@
 	}
 	if (((fileobject*)f)->f_fp == NULL)
 		return err_closed();
-	return getline((fileobject *)f, n);
+	return mygetline((fileobject *)f, n);
 }

 /* Python method */
@@ -507,7 +507,7 @@
 		}
 	}

-	return getline(f, n);
+	return mygetline(f, n);
 }

 static object *
@@ -525,7 +525,7 @@
 	if ((list = newlistobject(0)) == NULL)
 		return NULL;
 	for (;;) {
-		line = getline(f, 0);
+		line = mygetline(f, 0);
 		if (line != NULL && getstringsize(line) == 0) {
 			DECREF(line);
 			break;
